components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1052,9 +1074,96 @@ void ConversationDriver::PerformAssistantGeneration(
+            {base::MayBlock(), base::TaskPriority::BEST_EFFORT});
+    text_embedder_ = TextEmbedder::Create(
+        base::FilePath(universal_qa_model_path_), embedder_task_runner);
+  }
+  if (text_embedder_ && should_refine_page_content) {
+    if (text_embedder_->IsInitialized()) {
+      text_embedder_->GetTopSimilarityWithPromptTilContextLimit(
+          input, page_content,
+          GetCurrentModel()
+              .options->get_leo_model_options()
+              ->max_page_content_length,
+          base::BindOnce(&ConversationDriver::OnGetRefinedPageContent,
+                         weak_ptr_factory_.GetWeakPtr(), input,
+                         std::move(data_received_callback),
+                         std::move(data_completed_callback), page_content,
+                         is_video));
+    } else {
+      text_embedder_->Initialize(base::BindOnce(
+          &ConversationDriver::OnTextEmbedderInitialized,
+          weak_ptr_factory_.GetWeakPtr(), input,
+          std::move(data_received_callback), std::move(data_completed_callback),
+          std::move(page_content), is_video));
+    }
+  } else {
+    engine_->GenerateAssistantResponse(is_video, page_content, chat_history_,
+                                       input, std::move(data_received_callback),
+                                       std::move(data_completed_callback));
+  }
+}
+
+void ConversationDriver::OnTextEmbedderInitialized(
+    const std::string& input,
+    EngineConsumer::GenerationDataCallback data_received_callback,
+    EngineConsumer::GenerationCompletedCallback data_completed_callback,
+    std::string page_content,
+    bool is_video,
+    bool initialized) {
+  if (initialized) {
+    text_embedder_->GetTopSimilarityWithPromptTilContextLimit(
+        input, page_content,
+        GetCurrentModel()
+            .options->get_leo_model_options()
+            ->max_page_content_length,
+        base::BindOnce(&ConversationDriver::OnGetRefinedPageContent,
+                       weak_ptr_factory_.GetWeakPtr(), input,
+                       std::move(data_received_callback),
+                       std::move(data_completed_callback), page_content,
+                       is_video));
+  } else {
+    VLOG(1) << "Failed to initialize TextEmbedder";
+    engine_->GenerateAssistantResponse(
+        is_video, std::move(page_content), chat_history_, input,
+        std::move(data_received_callback), std::move(data_completed_callback));
+  }
+}
+void ConversationDriver::OnGetRefinedPageContent(
+    const std::string& input,
+    EngineConsumer::GenerationDataCallback data_received_callback,
+    EngineConsumer::GenerationCompletedCallback data_completed_callback,
+    std::string page_content,
+    bool is_video,
+    base::expected<std::string, std::string> refined_page_content) {
+  std::string page_content_to_use = page_content;
+  if (refined_page_content.has_value()) {
+    page_content_to_use = refined_page_content.value();
```

Comment:
`std::move()` strings please here and above

Reply:
fixed in 0e1761796c79c786edf00a9459ff806f8263a294